### PR TITLE
[MOBILE-1044] update iOS to SDK 13.0.1

### PR DESCRIPTION
**Pull request by crow**
Updates the react-native plugin to iOS SDK 13.0.1

### What do these changes do?
Migrates the plugin to work with iOS SDK 13.0.1

### How did you verify these changes?
Manual testing with the sample.
Tested inbox action, open inbox, opening messages, various other actions including landing pages just because.

#### Verification Screenshots:
react-native

### CODE REVIEW

File: UrbanAirship-React-Native-Module.podspec [hunk 1]
```diff
@@ -13,5 +13,6 @@ require "json"
   s.source       = { :git => "https://github.com/urbanairship/react-native-module.git", :tag => "{s.version}" }
   s.source_files  = "ios/**/*.{h,m}"
   s.dependency "React"
-  s.dependency "UrbanAirship-iOS-SDK", "12.1.2"
+  s.dependency "Airship", "13.0.1"
+  s.dependency "Airship/Location", "13.0.1"
```
Comment (rlepinski) [inline UrbanAirship-React-Native-Module.podspec]: ^ do not include location here

File: ios/UARCTModule/UARCTMessageView.m [hunk 2]
```diff
@@ -22,11 +22,13 @@ @implementation UARCTMessageView
 - (instancetype) init {
     self = [super initWithFrame:CGRectZero];
     if (self) {
-        self.nativeBridge = [[UAWKWebViewNativeBridge alloc] init];
-        self.nativeBridge.forwardDelegate = self;
+        self.nativeBridge = [UANativeBridge nativeBridge];
+        self.nativeBridge.forwardNavigationDelegate = self;
+        self.nativeBridge.nativeBridgeDelegate = self;
```
Comment (rlepinski) [inline ios/UARCTModule/UARCTMessageView.m]: Need at add message center extension to the native bridge:

```
    self.nativeBridgeExtension = [[UAMessageCenterNativeBridgeExtension alloc] init];
    self.nativeBridge.nativeBridgeExtensionDelegate = self.nativeBridgeExtension;
```

File: ios/UARCTModule/UARCTMessageCenter.m [hunk 3]
```diff
@@ -24,22 +24,40 @@ + (UARCTMessageCenter *)shared {
...
+            [[UAMessageCenter shared].defaultUI displayMessageCenterAnimated:animated];
+        } else {
+            [[UARCTEventEmitter shared] showInbox];
+        }
+     });
+
+}
+
+- (void)displayMessageCenterForMessageID:(NSString *)messageID animated:(BOOL)animated {
+    dispatch_async(dispatch_get_main_queue(), ^{
```
Comment (rlepinski) [inline ios/UARCTModule/UARCTMessageCenter.m]: Are these not running on the main queue?

File: ios/UARCTModule/UrbanAirshipReactModule.m [hunk 4]
```diff
@@ -399,45 +401,34 @@ - (RCTBridge *)bridge {
                  overlay:(BOOL)overlay
                  displayMessage_resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject) {
+    UARCTMessageViewController *mvc = [[UARCTMessageViewController alloc] initWithNibName:@"UAMessageCenterMessageViewController" bundle:[UAMessageCenterResources bundle]];
+    [mvc loadMessageForID:messageId onlyIfChanged:YES onError:nil];
 
-    if (overlay) {
```
Comment (marc-scig) [inline ios/UARCTModule/UrbanAirshipReactModule.m]: are we not handling this case anymore? I noticed the displayOverlayMessage method is gone as well.

Comment (marc-scig) [reply ios/UARCTModule/UrbanAirshipReactModule.m]: if removing it was intentional, does that mean we should update the api/docs on the js side?

File: ios/UARCTModule/UARCTMessageCenter.m (hunk 3 above)
Comment (crow) [reply ios/UARCTModule/UARCTMessageCenter.m]: this is just cruft I pulled in when going off what we do in the samples, removed

File: ios/UARCTModule/UrbanAirshipReactModule.m (hunk 4 above)
Comment (crow) [reply ios/UARCTModule/UrbanAirshipReactModule.m]: removed and updated both platforms and js interface/doc comments

**Pull request merged**